src/eckit/types/Time.cc

```diff
@@ -127,9 +147,9 @@ Time& Time::operator=(const Time& other) {
     return *this;
 }
 
-Time::Time(long hh, long mm, long ss) :
+Time::Time(long hh, long mm, long ss, bool extended) :
     seconds_(hh * 3600 + mm * 60 + ss) {
-    if (hh >= 24 || mm >= 60 || ss >= 60 || hh < 0 || mm < 0 || ss < 0) {
+    if (mm >= 60 || ss >= 60 || (!extended && (hh >= 24 || hh < 0 || mm < 0 || ss < 0))) {
```

Comment:
IS there a reason we can't don't support 61 minutes, or 61 seconds, and lroll them into larger time units, given that we do that with hours here?